crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1264,15 +1266,20 @@ impl<'db> TypeInferenceBuilder<'db> {
             .types
             .expression_ty(iterable.scoped_ast_id(self.db, self.scope));
 
+        tracing::debug!("For::IterableTy: {}", iterable_ty.display(self.db));
+        tracing::debug!("For::IsAsync: {}", is_async);
+
         let loop_var_value_ty = if is_async {
             // TODO(Alex): async iterables/iterators!
-            Type::Unknown
+            Type::Todo
         } else {
             iterable_ty
                 .iterate(self.db)
                 .unwrap_with_diagnostic(iterable.into(), self)
         };
 
+        tracing::debug!("For::LoopVar: {}", loop_var_value_ty.display(self.db));
```

Comment:
I think having all three of these would lead to very noisy tracing output. We should be able to easily figure out the inferred type of a `for` loop variable using `reveal_type`, so I'm not sure we really need any of them for debugging issues from users; I think I'd remove all three

Reply:
Sorry! That was also a temporary code I used to isolate the bug - not meant to be commited!